src/ProfileBundle/Block/AccountBlockService.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\ProfileBundle\Block;
+
+use Sonata\BlockBundle\Block\BlockContextInterface;
+use Sonata\BlockBundle\Block\Service\AbstractAdminBlockService;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\SecurityContextInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * Render a block with the connection option or the login name.
+ *
+ * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ */
+class AccountBlockService extends AbstractAdminBlockService
+{
+    /**
+     * @var TokenStorageInterface|SecurityContextInterface
+     */
+    private $tokenStorage;
+
+    /**
+     * NEXT_MAJOR: Go back to type hinting check when bumping requirements to SF 2.6+.
+     *
+     * @param string                                         $name
+     * @param TokenStorageInterface|SecurityContextInterface $tokenStorage
+     */
+    public function __construct($name, EngineInterface $templating, $tokenStorage)
+    {
```

Comment:
Maybe add the type hint? Like the comment mentions:
'NEXT_MAJOR: Go back to type hinting check when bumping requirements to SF 2.6+.'

Reply:
I miss it. SecurityContextInterface should be deleted becouse ecommere require symfony 3.4 || 4.x